MusicApp.PlaylistService.Application/Services/Implementations/SongService.cs

```diff
@@ -0,0 +1,40 @@
+﻿using AutoMapper;
+using MusicApp.PlaylistService.Application.DTOs;
+using MusicApp.PlaylistService.Application.Repositories;
+using MusicApp.PlaylistService.Application.Services.Interfaces;
+using MusicApp.PlaylistService.Domain.Entities;
+
+namespace MusicApp.PlaylistService.Application.Services.Implementations;
+
+public class SongService : ISongService
+{
+    private readonly ISongRepository _songRepository;
+    private readonly IUserRepository _userRepository;
+    private readonly IMapper _mapper;
+
+    public SongService(ISongRepository songRepository, IUserRepository userRepository, IMapper mapper)
+    {
+        _songRepository = songRepository;
+        _userRepository = userRepository;
+        _mapper = mapper;
+    }
+
+    public async Task<IEnumerable<SongOutputDto>> UpdateSongs(IEnumerable<Song> songs, CancellationToken cancellationToken)
+    {
+        await _songRepository.DeleteAllSongs(cancellationToken);
+
+        foreach(var song in songs)
```

Comment:
Are there any reasons to call SaveChanges in every iteration of loop?

Reply:
I forgot to delete this service. Dasha told me that it's better to send a message right after creating/updating/deleting a song. Therefore, this method and service is no longer needed.